src/Elastic.Markdown/Myst/Components/ApplicableToComponent.cshtml

```diff
@@ -111,209 +111,212 @@
-					}
 
-					break;
-				case ProductLifecycle.Beta:
-					if (TryGetRealVersion(applicability, out var betaVersion) && betaVersion > versioningSystem.Current)
-					{
-						badgeText = "Planned";
-						lifecycleTooltip = "We plan to add this functionality in a future update. Plans may change without notice.";
-						lifecycleClass = "planned";
-					}
-					else
+				if (realVersion <= versioningSystem.Current)
+				{
+					tooltipText = $"{lifecycleFull} in {keyFull} in version {realVersion} and later unless otherwise specified.";
+				}
+				else
+				{
+					if (applicability.Lifecycle is ProductLifecycle.GenerallyAvailable or ProductLifecycle.Beta or ProductLifecycle.TechnicalPreview)
 					{
-						lifecycleTooltip = "This functionality is in beta and is subject to change. The design and code is less mature than official GA features and is being provided as-is with no warranties. Beta features are not subject to the support SLA of official GA features.";
+						tooltipText = $"We plan to add this in a future {keyFull} update. Plans may change without notice.";
 					}
-
-					break;
-				case ProductLifecycle.GenerallyAvailable:
-					if (TryGetRealVersion(applicability, out var version) && version > versioningSystem.Current)
+					else if (applicability.Lifecycle is ProductLifecycle.Deprecated)
 					{
-						badgeText = "Planned";
-						lifecycleTooltip = "We plan to add this functionality in a future update. Plans may change without notice.";
-						lifecycleClass = "planned";
+						tooltipText = $"We plan to deprecate this in a future {keyFull} update. Plans may change without notice.";
 					}
-
-					break;
-				case ProductLifecycle.Deprecated:
-					if (TryGetRealVersion(applicability, out var deprecatedVersion) && deprecatedVersion > versioningSystem.Current)
+					else if (applicability.Lifecycle is ProductLifecycle.Removed)
 					{
-						badgeText = "Deprecation planned";
-						lifecycleTooltip = "We plan to deprecate this functionality in a future update. Plans may change without notice.";
+						tooltipText = $"We plan to remove this in a future {keyFull} update. Plans may change without notice.";
 					}
+				}
+				
+			}
+			else
+			{
+				tooltipText = $"{lifecycleFull} in {keyFull} unless otherwise specified.";
+			}
 
-					break;
-				case ProductLifecycle.Removed:
-					if (TryGetRealVersion(applicability, out var removedVersion) && removedVersion > versioningSystem.Current)
-					{
-						badgeText = "Removal planned";
-						lifecycleTooltip = "We plan to remove this functionality in a future update. Plans may change without notice.";
-					}
+			var disclaimer = applicability.Lifecycle switch
+			{
+				ProductLifecycle.Beta => "Beta features are subject to change. The design and code is less mature than official GA features and is being provided as-is with no warranties. Beta features are not subject to the support SLA of official GA features.",
```

Comment:
So this could be something like:

```
// lifecycleFull = "Available in beta"
// keyFull = "Elastic Stack"
// realVersion = "9.1.0"

= "Available in beta in Elastic Stack in version 9.1.0 and later unless otherwise specified." +
"Beta features are subject to change. The design and code is less mature than official GA features
and is being provided as-is with no warranties. Beta features are not subject to the support SLA of
official GA features."

```

Reply:
This is a bit dense: `Available in beta in Elastic Stack in version 9.1.0 and later unless otherwise specified.` I wonder if we can tighten it up a bit.

Reply:
Available in beta **for** Elastic Stack ~in~ version 9.1.0 and later unless otherwise specified?

Reply:
Spoke with Florent this morning about this. We can move forward with this PR and later make any additional updates that come out https://github.com/elastic/docs-content-internal/issues/28.